src/onCLS.ts

```diff
@@ -85,12 +87,10 @@ export const onCLS = (
           opts!.reportAllChanges,
         );
 
-        document.addEventListener('visibilitychange', () => {
-          if (document.visibilityState === 'hidden') {
-            handleEntries(po.takeRecords() as CLSMetric['entries']);
-            report(true);
-          }
-        });
+        visibilityWatcher.onHidden = () => {
```

Comment:
Ahh, I see you're using it as a setter here as well.

I'd make this a regular function that takes a callback, otherwise it looks like setting a function overrides the previously set value (similar to the native event handler properties like `.onload = `).